tools/kubernetes/shared-cluster.sh

```diff
@@ -0,0 +1,149 @@
+        --region="$GKE_REGION" \
+        --machine-type=n1-standard-8 \
+        --scopes=storage-full # --accelerator type="$GKE_GPU_TYPE",count="$GKE_GPU_PER_NODE"
+
+    gcloud compute instances create "$BASTION_INSTANCE_NAME" \
+        --project "$GCP_PROJECT_NAME" \
+        --zone="$BASTION_INSTANCE_ZONE" \
+        --machine-type="$BASTION_INSTANCE_TYPE" \
+        --network-interface=no-address,network-tier=PREMIUM,subnet="$GCS_SUBNET_NAME"
+
+elif [ ""$ACTION"" = "connect" ]; then
+    set -e
+    # Kill old tunnels.
+    # For some reason setting:
+    #   sudo sh -c 'echo "StreamLocalBindUnlink yes" >> /etc/ssh/sshd_config'
+    # on the server didn't work.
+    gcloud compute ssh "$BASTION_INSTANCE_NAME" \
+        --project "$GCP_PROJECT_NAME" \
+        --zone="$BASTION_INSTANCE_ZONE" \
+        --tunnel-through-iap \
+        -- pkill -u '$USER' -x -f '"^sshd: $USER[ ]*$"' || true
+
+    export MASTER_IP_FROM_INTERNAL=$(
+        gcloud compute instances describe "$BASTION_INSTANCE_NAME" \
+            --zone "$BASTION_INSTANCE_ZONE" \
+            --project "$GCP_PROJECT_NAME" \
+            | yq -o=json \
+            | jq '.networkInterfaces[0].networkIP'
+    )
+    export MASTER_PORT_FROM_INTERNAL=$(jot -r 1 2000 65000)
+
+    export SOCKS5_PROXY_PORT=60001 # Exported for envsubst call. Same for anything else exported.
+    SOCKS5_PROXY_TUNNEL_OPTS="-D "$SOCKS5_PROXY_PORT" -nNT"
+    gcloud compute ssh "$BASTION_INSTANCE_NAME" \
+        --project "$GCP_PROJECT_NAME" \
+        --zone="$BASTION_INSTANCE_ZONE" \
+        --tunnel-through-iap \
+        --ssh-flag="-4 -C -NR$MASTER_PORT_FROM_INTERNAL:localhost:$MASTER_PORT_FROM_INTERNAL $SOCKS5_PROXY_TUNNEL_OPTS" &
+    TUNNEL_PID=$!
+    trap 'kill $TUNNEL_PID' EXIT
+    export SOCKS5_PROXY_URL="socks5://localhost:$SOCKS5_PROXY_PORT"
+
+    # kubectl won't respect ALL_PROXY, only HTTPS_PROXY.
+    # https://kubernetes.io/docs/tasks/extend-kubernetes/socks5-proxy-access-api/#client-configuration
+    echo "Cluster is setup, please use HTTPS_PROXY=$SOCKS5_PROXY_URL when using kubectl"
+
+    gcloud container clusters get-credentials "$GKE_CLUSTER_NAME" \
+        --region="$GKE_REGION" \
+        --project="$GCP_PROJECT_NAME"
+
+    export ALL_PROXY=$SOCKS5_PROXY_URL HTTP_PROXY=$SOCKS5_PROXY_URL HTTPS_PROXY=$SOCKS5_PROXY_URL
+    export KUBERNETES_NAMESPACE=$USER-devcluster
+    for i in 1 2 3 4 5; do
+        kubectl create namespace $KUBERNETES_NAMESPACE --dry-run=client -o yaml | kubectl apply -f - && break \
+            || echo "Trying to create namespace $KUBERNETES_NAMESPACE for you again in 2 seconds..." \
+            && sleep 2
+
+    done
+
+    # Although devcluster supports variables, numeric values fail to load, so
+    # Manually apply those into a temp file.
+    TMPYAML=/tmp/devcluster-kubernetes.yaml
+    rm -f $TMPYAML
+    envsubst <devcluster.tpl.yaml >$TMPYAML
+
+    export ALL_PROXY=$SOCKS5_PROXY_URL
+    devcluster --oneshot --config $TMPYAML
```

Comment:
Since the `devcluster` command (without the `cwd` flag) assumes that we are in the `determined` directory and the readMe commands imply that we run this script from `determined/tools/kubernetes`, I think we might need to either add the flag so that we run `devcluster --oneshot --config $TMPYAML --cwd ../..` or change the readMe to execute `./tools/kubernetes/shared-cluster.sh up/connect/down` in our script instructions.